src/components/AppNavi.vue

```diff
@@ -294,18 +294,13 @@ export default {
                 t('cookbook', 'Path to your recipe collection'),
                 (path) => {
                     let $this = this
-                    axios({
-                        url: this.$window.baseUrl + '/config',
-                        method: 'POST',
-                        data: { 'folder': path },
-                    })
+                    this.$store.dispatch('updateRecipeDirectory', { dir: path })
                         .then(() => {
-                            $this.loadAll()
-                            .then(() => {
-                                $this.$store.dispatch('setRecipe', { recipe: null })
-                                $this.$window.goTo('/')
-                                $this.recipeFolder = path
-                            })
+                            $this.recipeFolder = path
+                            $this.$root.$emit('refreshNavigation')
```

Comment:
Why emitting on the `$root` element?

Reply:
The `$root` element is used as a global bus here

Reply:
While this is in principal okay, I still changed this a bit